docs/src/crypto-primitives/tsmt.md

```diff
@@ -0,0 +1,123 @@
+# Tiered Sparse Merkle Tree (TSMT)
+A sparse Merkle tree (SMT) is a cryptographic data structure that allows authenticated manipulations of a key-value store i.e. a dictionary. It is a Merkle tree because it is
```

Comment:
I think there should be a `,` before the `i.e.`. 

https://jakubmarian.com/comma-after-i-e-and-e-g/